lib/phoenix_live_view/html_tokenizer.ex

```diff
@@ -294,6 +308,12 @@ defmodule Phoenix.LiveView.HTMLTokenizer do
     end
   end
 
+  defp classify_tag_type(":" <> _name), do: {:ok, :slot}
+  defp classify_tag_type(":inner_block"), do: {:error, "the slot name :inner_block is reserved"}
+  defp classify_tag_type(<<first, _::binary>>) when first in ?A..?Z, do: {:ok, :remote_component}
+  defp classify_tag_type("." <> _name), do: {:ok, :local_component}
+  defp classify_tag_type(_name), do: {:ok, :tag}
```

Comment:
I noticed one issue. We are still checking for `":"` and `"."` in the engine. This means the engine is still aware of the slot and other conventions. I think the return type has to actually be this:

```elixir
{:slot, slot_name_as_atom}
{:remote_component, Module, function_as_atom}
{:local_component, function_as_atom}
:tag
{:error, reason_as_string}
```

This way the engine doesn't have to check for `.` or `:` and so on.

WDYT?

Reply:
Yeah, that was bothering me to. I will try to improve that based on your comment. Thanks

Reply:
after the last commit, this is the only part we check for `.` at HTMLEngine. I guess that's fine since there we actually want to get the function name from the module. Let me know if you see an opportunity to refactor this.